freqtrade/strategy/interface.py

```diff
@@ -655,12 +655,20 @@ def should_sell(self, trade: Trade, rate: float, date: datetime, buy: bool,
         # Stoploss
         if roi_reached and stoplossflag.sell_type != SellType.STOP_LOSS:
             logger.debug(f"{trade.pair} - Required profit reached. sell_type=SellType.ROI")
+            if hasattr(trade, 'buy_tag') and trade.buy_tag is not None:
```

Comment:
You'll not have to test with `hasattr` here. 
Trade is an internal class - where old versions of freqtrade won't have it - but new versions do.
As such - it's safe to assume it's always there as attribute (although it can obviously be empty).